src/analysis/special_functions/gamma.lean

```diff
@@ -0,0 +1,587 @@
+/-
+Copyright (c) 2022 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+
+import analysis.special_functions.exponential
+import measure_theory.integral.integral_eq_improper
+import measure_theory.integral.limit_comparison
+import analysis.special_functions.integrals
+
+/-!
+# The Gamma function
+
+In this file we define the Γ function (of a real variable in the range `1 ≤ s`), using the
+definition `Γ(s) = ∫ x in 0..∞, x^(s-1) exp(-x) dx`, and prove that it is continuous and satisfies
+the relation `Γ(s+1) = s Γ(s)`. We also prove that `Γ(n+1) = n!` for `n ∈ ℕ`.
+
+TO DO:
+
+- Extend to the whole real line.
+- Allow complex `s` and prove analyticity.
+
+## Tags
+
+gamma
+-/
+
+noncomputable theory
+open finset filter metric interval_integral set function
+open_locale classical topological_space ennreal filter measure_theory
+
+namespace real
+
+/-- Integrand for the Gamma function integral -/
+def F (s x : ℝ) : ℝ := exp(-x) * x^s
```

Comment:
This needs a good name, at the moment it will be in the global namespace as `real.F` which is a bit too generic. I don't know a name for this so maybe `gamma_integrand` will do?
You can add a local notation in the file so you don't have to write the full name everywhere if you want. But the names of lemmas referring to this will also need to be renamed to whatever you decide to call `F`.